annnufan/assignment1/solution_1.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <algorithm>
+#include <cctype>
+#include <functional>
+
+void word_anagram(bool case_sens) {
+	std::string word_1, word_2;
+	std::cin >> word_1 >> word_2;

```

Comment:
You should write your code in a way that it is portable and ready to be used as part of a larger system.
Think of anyone trying to use this code in a different context:
What if the input is not from the keyboard, but from a file, from an internet database or any other source?
What if they want to use the return value as part of a backend system, and not only print it to the screen?
In that case, the best way to expose this code is using a function, for example:
```C++
bool word_anagram(string word_1, string word_2, bool case_sens);
```
(The return value can hold whether the input words are anagrams or not).

For this reason, always try to separate the user input/output and the actual assignment logic.
For example, you can wrap this function later with the user input and output:
```C++
std::string word_1, word_2;
std::cin >> word_1 >> word_2;
if (word_anagram(word_1, word_2, case_sens)) {
  cout << "anagrams";
} else {
  cout << "not anagrams";
}